conans/tools.py

```diff
@@ -165,6 +165,17 @@ def build_sln_command(settings, sln_path, targets=None, upgrade_project=True, bu
         command += " /target:%s" % ";".join(targets)
     return command
 
+def vs_installation_path(version):
+    if not hasattr(vs_installation_path, "_cached"):
```

Comment:
generally speaking, where are many cases where it can fail:
- os.environ fails because where is no ProgramFiles(x86)/ProgramFiles environment variables for some reason (e.g. someone trying to build for VS on non-Windows?)
- wswhere.exe doesn't exists at all
- subprocess call fail (e.g. there is no requested version of Visual Studio installed)
- decode can fail (subprocess output cannot be represented in current system encoding)
so, @lasote, what shall I do, handle all possible exceptions and return None for them all?